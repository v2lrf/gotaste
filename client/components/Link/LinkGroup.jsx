import React from 'react'
import PropTypes from 'prop-types'

function LinkGroup({ label, children }) {
  return (
    <div className="text-right mb-4">
      {label && (
        <span className="mr-2 text-grey-dark text-sm font-bold uppercase">
          {label}
        </span>
      )}
      {children}
    </div>
  )
}

LinkGroup.propTypes = {
  label: PropTypes.string,
  children: PropTypes.node.isRequired
}

LinkGroup.defaultProps = {
  label: null
}

export default LinkGroup
