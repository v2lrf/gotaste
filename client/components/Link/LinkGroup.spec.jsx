import React from 'react'
import LinkGroup from './LinkGroup'

describe('LinkGroup', () => {
  let component
  beforeEach(() => {
    component = shallow(<LinkGroup>Some content!</LinkGroup>)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    it('should match a snapshot with a label', () => {
      component.setProps({ label: 'Some label' })
      expect(component).toMatchSnapshot()
    })
  })
})
