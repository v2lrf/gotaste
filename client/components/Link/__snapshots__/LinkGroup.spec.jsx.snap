// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkGroup Snapshots should match a snapshot with a label 1`] = `
<div
  className="text-right mb-4"
>
  <span
    className="mr-2 text-grey-dark text-sm font-bold uppercase"
  >
    Some label
  </span>
  Some content!
</div>
`;

exports[`LinkGroup Snapshots should match the default snapshot 1`] = `
<div
  className="text-right mb-4"
>
  Some content!
</div>
`;
